core/modules/node/node.pages.inc

```diff
@@ -458,6 +515,209 @@ function node_form_submit($form, &$form_state) {
+  }
+
+  $node->changed = REQUEST_TIME;
+ 
+  // Get the view mode to render the preview in.
+  $view_mode = empty($_GET['view_mode']) ? 'full' : $_GET['view_mode'];
+
+  // Property so we can manipulate $page in template_preprocess_node.
+  if ($view_mode == 'full' || $view_mode == 'default') {
+    backdrop_set_title($node->title);
+    $node->node_page_title = TRUE;
+  }
+ 
+  $build = array();
+  // @todo all links should be disabled, except for returning to the edit form.
+  $build['#attached']['js'][] = backdrop_get_path('module', 'node') . '/js/node.preview.js';
+  $build['#attached']['css'][] = backdrop_get_path('module', 'node') . '/css/node.preview.css';
+  $build['preview_form_select'] = array(
+    '#type' => 'container',
+    '#attributes' => array(
+      'class' => array('node-preview-container', 'container-inline')
+    ),
+  );
+
+  $form = backdrop_get_form('node_preview_banner_form', $node, $node_tempstore_id);
+  $build['preview_form_select']['view-mode'] = $form;
+  $build['preview'] = node_view($node, $view_mode);
+  return $build;
+}
+
+/**
+ * Get the preview form selection box.
+ *
+ * @param $node
+ *   A node entity.
+ * @param $node_tempstore_id
+ *   The tempstore ID of the node being previewed.
+ *
+ * @return array $form
+ *   The form which provides access to node preview operations.
+ */
+function node_preview_banner_form(array $form, array $form_state, Node $node, $node_tempstore_id) {
+  // Always add the 'default' view mode.
+  $view_mode_options = array();
+
+  // Get view mode options.
+  $entity_info = entity_get_info('node');
+  $view_modes = $entity_info['view modes'];
+  if (is_array($view_modes)) {
+    backdrop_sort($view_modes, array('label' => SORT_STRING));
+  }
+
+  foreach ($view_modes as $key => $info) {
+    if (!in_array($key, array('rss', 'search_index', 'search_result'))) {
+      $view_mode_options[$key] = $info['label'];
+    }
+
+    // In case this is the 'full' view mode, rename default.
+    if ($key == 'full') {
+      $view_mode_options[$key] = t('Default');
+    }
+  }
+
+  $path = 'node/add/' . $node->bundle();
+  if (!empty($node->nid)) {
+    $path = 'node/' . $node->nid . '/edit';
+  }
+  $query = array('node_tempstore_id' => $node_tempstore_id);
+  $url = url($path, array('query' => $query, 'absolute' => TRUE));
+
+  // Get the view mode to render the preview in.
+  $view_mode = empty($_GET['view_mode']) ? 'full' : $_GET['view_mode'];
+
+  $form['node'] = array(
+    '#type' => 'value',
+    '#value' => $node,
+  );
+
+  $form['node_tempstore_id'] = array(
+    '#type' => 'value',
+    '#value' => $node_tempstore_id,
+  );
+
+  $form['backlink'] = array(
+    '#type' => 'link',
+    '#title' => t('Back to content editing'),
+    '#href' => $url,
+    '#attributes' => array('class' => array('node-preview-backlink')),
+  );
+
+  $form['view_mode'] = array(
+    '#type' => 'select',
+    '#title' => t('Select a display mode'),
+    //'#title_display' => 'invisible',
```

Comment:
Probably remove this if we're not going to use it.

Reply:
done